clients/lsp-eslint.el

```diff
@@ -260,27 +260,45 @@ stored."
-                                                    :mode lsp-eslint-fix-all-problem-type)
-                            :format (lsp-json-bool lsp-eslint-format)
-                            :quiet (lsp-json-bool lsp-eslint-quiet)
-                            :onIgnoredFiles (if lsp-eslint-warn-on-ignored-files "warn" "off")
-                            :options (or lsp-eslint-options (ht))
-                            :rulesCustomizations lsp-eslint-rules-customizations
-                            :run lsp-eslint-run
-                            :nodePath lsp-eslint-node-path
-                            :workspaceFolder (list :uri (lsp--path-to-uri workspace-folder)
-                                                   :name (f-filename workspace-folder)))))))
+                      (let ((working-directory (lsp--working-directory workspace-folder file)))
+                        (list :validate (if (member (lsp-buffer-language) lsp-eslint-validate) "on" "probe")
+                              :packageManager lsp-eslint-package-manager
+                              :codeAction (list
+                                           :disableRuleComment (list
+                                                                :enable (lsp-json-bool lsp-eslint-code-action-disable-rule-comment)
+                                                                :location lsp-eslint-code-action-disable-rule-comment-location)
+                                           :showDocumentation (list
+                                                               :enable (lsp-json-bool lsp-eslint-code-action-show-documentation)))
+                              :codeActionOnSave (list :enable (lsp-json-bool lsp-eslint-auto-fix-on-save)
+                                                      :mode lsp-eslint-fix-all-problem-type)
+                              :format (lsp-json-bool lsp-eslint-format)
+                              :quiet (lsp-json-bool lsp-eslint-quiet)
+                              :onIgnoredFiles (if lsp-eslint-warn-on-ignored-files "warn" "off")
+                              :options (or lsp-eslint-options (ht))
+                              :rulesCustomizations lsp-eslint-rules-customizations
+                              :run lsp-eslint-run
+                              :nodePath lsp-eslint-node-path
+                              :workingDirectory (when working-directory
+                                                  (list
+                                                   :directory working-directory
+                                                   :!cwd :json-false))
+                              :workspaceFolder (list :uri (lsp--path-to-uri workspace-folder)
+                                                     :name (f-filename workspace-folder))))))))
        (apply #'vector)))
 
+(defun lsp--working-directory (workspace current-file)
```

Comment:
can you prefix with `lsp-eslint--`? Can you also drop one line what this method is supposed to do.

Reply:
Done